Content.Shared/Health/BodySystem/BodyManagerComponent.cs

```diff
@@ -0,0 +1,275 @@
+﻿using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using System;
+using System.Collections.Generic;
+using Content.Shared.BodySystem;
+using Robust.Shared.ViewVariables;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Map;
+using System.Linq;
+
+namespace Content.Shared.BodySystem {
+    public enum BodyPartCompatibility { Mechanical, Biological, Universal };
+    public enum BodyPartType { Other, Torso, Head, Arm, Hand, Leg, Foot };
+    public enum SurgeryToolType { Incision, Retraction, Cauterization }
+
+
+}
+
+namespace Content.Shared.BodySystem {
+
+    /// <summary>
+    ///     Component representing the many BodyParts attached to each other. 
+    /// </summary>
+    [RegisterComponent]
+    public class BodyManagerComponent : Component {
+
+#pragma warning disable CS0649
+        [Dependency]
+        private IPrototypeManager _prototypeManager;
+#pragma warning restore
+
+        [ViewVariables]
+        private BodyTemplate _template;
+        [ViewVariables]
+        private Dictionary<string, BodyPart> _partDictionary = new Dictionary<string, BodyPart>();
+
+        public sealed override string Name => "BodyManager";
+        /// <summary>
+        ///     The BodyTemplate that this BodyManagerComponent is adhering to.
+        /// </summary>
+        public BodyTemplate Template => _template;
+        /// <summary>
+        ///     Maps BodyTemplate slot name to the BodyPart object filling it (if there is one). 
+        /// </summary>
+        public Dictionary<string, BodyPart> PartDictionary => _partDictionary;
+        /// <summary>
+        ///     List of all BodyParts in this body, taken from the keys of _parts.
+        /// </summary>
+        public List<BodyPart> Parts {
```

Comment:
It's cleaner to return an `IEnumerable<BodyPart>` here. It's more performant and cleaner.

You can then just return `_partDictionary.Values` and don't have to deal with manually copying it into a list.